ospfd/ospf_zebra.c

```diff
@@ -396,6 +397,101 @@ struct ospf_external *ospf_external_add(struct ospf *ospf, uint8_t type,
 	return ext;
 }
 
+/*
+ * Walk all the ei received from zebra for a route type and apply
+ * default route-map.
+ */
+bool ospf_external_default_routemap_apply_walk(struct ospf *ospf,
+					       struct list *ext_list,
+					       struct external_info *default_ei)
+{
+	struct listnode *node;
+	struct ospf_external *ext;
+	struct route_node *rn;
+	struct external_info *ei;
+	int ret;
+
+	for (ALL_LIST_ELEMENTS_RO(ext_list, node, ext)) {
+		if (!ext->external_info)
+			continue;
+
+		for (rn = route_top(ext->external_info); rn;
+		     rn = route_next(rn)) {
+			if ((ei = rn->info) != NULL) {
```

Comment:
```
ei = rn->info;
if (!ei)
    continue;
```

That would save you one indentation level, then you can move the zlog_debug() here and fix the build error.